docs/user_guide/representing/array.md

```diff
@@ -439,6 +439,104 @@ assert not my_doc.is_view()  # False
     To summarize: you should use `DocVec` when you need to work with contiguous data, and you should use `DocList` when you need to rearrange
     or extend your data.
 
+
+## Dealing with Optional Field
+
+
+Both [`DocList`][docarray.array.doc_list.doc_list.DocList] and [`DocVec`][docarray.array.doc_vec.doc_vec.DocVec] support optional fields but they behave differently.
+
+!!! note
+    This whole section is specific for nested BaseDoc, DocList and DocVec. 
+    for other data type DocList and DocVec will treat the optional case as a normal case.
+
+Let's take an example:
+
+```python
+from typing import Optional
+from docarray.typing import NdArray
+import numpy as np
+
+
+class ImageDoc(BaseDoc):
+    tensor: NdArray
+
+
+class ArticleDoc(BaseDoc):
+    image: Optional[ImageDoc]
+    title: str
+```
+
+In this example `ArticleDoc` has an optional field `image` which is an `ImageDoc`. This means that this field can either
+be None or be a `ImageDoc` instance.
+
+Remember for both DocList and DocVec calling `docs.image` will return a list like object of all the images of the documents.
+
+For DocList it will just iterate over all the documents and collect the image attribute of each document in a sequence for DocVec it will return the column of the image attribute.
```

Comment:
```suggestion
For DocList this call will iterate over all the documents and collect the image attribute of each document in a sequence, and for DocVec it will return the already stacked column of the `.image` attribute.
```